File: 백준/Silver/2606. 바이러스/바이러스.py
N = int(input())
M = int(input())

graph = [[] for _ in range(N+1)]
visited = [False] * (N+1)
for i in range(M):
    a,b = map(int,input().split())
    graph[a].append(b)
    graph[b].append(a)

counter = 0

def dfs(v):
    global counter
    visited[v] = True
    for i in graph[v]:
        if not visited[i]:
            counter += 1
            dfs(i)
dfs(1)
print(counter)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;
import java.io.*;

public class Main{

    static int[][] graph;
    static boolean[] visited;

    static int N;
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        int K = Integer.parseInt(br.readLine());
        graph = new int[N+1][N+1];
        visited = new boolean[N+1];
        for(int i=0; i<K; i++){
           StringTokenizer st = new StringTokenizer(br.readLine());
           int a = Integer.parseInt(st.nextToken());
           int b = Integer.parseInt(st.nextToken());
           graph[a][b] = 1;
           graph[b][a] = 1;
        }
        dfs(1);
        int count = 0;
        for(int i=1; i<N+1; i++){
            if(visited[i]){
                count++;
            }
        }
        System.out.println(count - 1);

    }

    static void dfs(int node){
        visited[node] = true;
        for(int i=1; i< N+1; i++ ){
            if(graph[node][i] == 1 && !visited[i]){
                dfs(i);
            }
        }
    }
}